From: Darik Horn <dajhorn@vanadac.com>
Date: Wed, 15 Oct 2014 05:32:35 -0500
Subject: Remove the upstream udev component.

The 90-zfs rule is problematic because modprobe has side-effects that cramp
event-driven pool and dataset management. Remove the entire udev subtree for
merge convenience.

Instead, create a similar dh_installudev rule in the packaging overlay in a
subsequent commit.
---
 Makefile.am                   |  2 +-
 configure.ac                  |  2 --
 udev/Makefile.am              |  1 -
 udev/rules.d/.gitignore       |  4 ----
 udev/rules.d/60-zvol.rules.in |  6 ------
 udev/rules.d/69-vdev.rules.in |  9 ---------
 udev/rules.d/90-zfs.rules.in  | 12 ------------
 udev/rules.d/Makefile.am      | 20 --------------------
 8 files changed, 1 insertion(+), 55 deletions(-)
 delete mode 100644 udev/Makefile.am
 delete mode 100644 udev/rules.d/.gitignore
 delete mode 100644 udev/rules.d/60-zvol.rules.in
 delete mode 100644 udev/rules.d/69-vdev.rules.in
 delete mode 100644 udev/rules.d/90-zfs.rules.in
 delete mode 100644 udev/rules.d/Makefile.am

diff --git a/Makefile.am b/Makefile.am
index c6b0523..d6cc460 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -7,7 +7,7 @@ include $(top_srcdir)/config/tgz.am
 
 SUBDIRS = include rpm
 if CONFIG_USER
-SUBDIRS += udev etc man scripts lib cmd
+SUBDIRS += etc man scripts lib cmd
 endif
 if CONFIG_KERNEL
 SUBDIRS += module
diff --git a/configure.ac b/configure.ac
index 6ebca65..4a95182 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,8 +57,6 @@ ZFS_AC_DEBUG_DMU_TX
 
 AC_CONFIG_FILES([ 
 	Makefile
-	udev/Makefile
-	udev/rules.d/Makefile
 	etc/Makefile
 	etc/zfs/Makefile
 	man/Makefile
diff --git a/udev/Makefile.am b/udev/Makefile.am
deleted file mode 100644
index f930941..0000000
--- a/udev/Makefile.am
+++ /dev/null
@@ -1 +0,0 @@
-SUBDIRS = rules.d
diff --git a/udev/rules.d/.gitignore b/udev/rules.d/.gitignore
deleted file mode 100644
index e7f7be8..0000000
--- a/udev/rules.d/.gitignore
+++ /dev/null
@@ -1,4 +0,0 @@
-69-vdev.rules
-60-zpool.rules
-60-zvol.rules
-90-zfs.rules
diff --git a/udev/rules.d/60-zvol.rules.in b/udev/rules.d/60-zvol.rules.in
deleted file mode 100644
index 199f77e..0000000
--- a/udev/rules.d/60-zvol.rules.in
+++ /dev/null
@@ -1,6 +0,0 @@
-# Persistent links for zvol
-#
-# persistent disk links: /dev/zvol/dataset_name
-# also creates compatibilty symlink of /dev/dataset_name
-
-KERNEL=="zd*" SUBSYSTEM=="block" ACTION=="add|change" PROGRAM="@udevdir@/zvol_id $tempnode" SYMLINK+="zvol/%c %c"
diff --git a/udev/rules.d/69-vdev.rules.in b/udev/rules.d/69-vdev.rules.in
deleted file mode 100644
index 5c2940a..0000000
--- a/udev/rules.d/69-vdev.rules.in
+++ /dev/null
@@ -1,9 +0,0 @@
-#
-# @udevdir@/rules.d/69-vdev.rules
-#
-
-ENV{DEVTYPE}=="disk", IMPORT{program}="@udevdir@/vdev_id -d %k"
-
-KERNEL=="*[!0-9]", ENV{SUBSYSTEM}=="block", ENV{ID_VDEV}=="?*", SYMLINK+="$env{ID_VDEV_PATH}"
-KERNEL=="*[0-9]", ENV{SUBSYSTEM}=="block", ENV{DEVTYPE}=="partition", ENV{ID_VDEV}=="?*", SYMLINK+="$env{ID_VDEV_PATH}-part%n"
-KERNEL=="dm-[0-9]*", ENV{SUBSYSTEM}=="block", ENV{ID_VDEV}=="?*", SYMLINK+="$env{ID_VDEV_PATH}"
diff --git a/udev/rules.d/90-zfs.rules.in b/udev/rules.d/90-zfs.rules.in
deleted file mode 100644
index a2715d2..0000000
--- a/udev/rules.d/90-zfs.rules.in
+++ /dev/null
@@ -1,12 +0,0 @@
-SUBSYSTEM!="block|misc", GOTO="zfs_end"
-ACTION!="add|change", GOTO="zfs_end"
-
-ENV{ID_FS_TYPE}=="zfs", RUN+="/sbin/modprobe zfs"
-ENV{ID_FS_TYPE}=="zfs_member", RUN+="/sbin/modprobe zfs"
-
-KERNEL=="null", SYMLINK+="root"
-SYMLINK=="null", SYMLINK+="root"
-
-SUBSYSTEM=="misc", KERNEL=="zfs", RUN+="@sbindir@/zpool list"
-
-LABEL="zfs_end"
diff --git a/udev/rules.d/Makefile.am b/udev/rules.d/Makefile.am
deleted file mode 100644
index 6816add..0000000
--- a/udev/rules.d/Makefile.am
+++ /dev/null
@@ -1,20 +0,0 @@
-udevrule_DATA = \
-	$(top_srcdir)/udev/rules.d/69-vdev.rules \
-	$(top_srcdir)/udev/rules.d/60-zvol.rules \
-	$(top_srcdir)/udev/rules.d/90-zfs.rules
-
-EXTRA_DIST = \
-	$(top_srcdir)/udev/rules.d/69-vdev.rules.in \
-	$(top_srcdir)/udev/rules.d/60-zvol.rules.in \
-	$(top_srcdir)/udev/rules.d/90-zfs.rules.in
-
-$(udevrule_DATA):
-	-$(SED) -e 's,@bindir\@,$(bindir),g' \
-		-e 's,@sbindir\@,$(sbindir),g' \
-		-e 's,@udevdir\@,$(udevdir),g' \
-		-e 's,@udevruledir\@,$(udevruledir),g' \
-		-e 's,@sysconfdir\@,$(sysconfdir),g' \
-		'$@.in' >'$@'
-
-distclean-local::
-	-$(RM) $(udevrule_DATA)
